config/fcos/v1_6_exp/translate.go

```diff
@@ -367,3 +374,79 @@ func buildGrubConfig(gb Grub) string {
 	superUserCmd := fmt.Sprintf("set superusers=\"%s\"\n", strings.Join(allUsers, " "))
 	return "# Generated by Butane\n\n" + superUserCmd + strings.Join(cmds, "\n") + "\n"
 }
+
+func (c Config) handleSelinux(options common.TranslateOptions) (types.Config, translate.TranslationSet, report.Report) {
+	rendered := types.Config{}
+	ts := translate.NewTranslationSet("yaml", "json")
+	var r report.Report
+
+	for i, module := range c.Selinux.Module {
+		yamlPath := path.New("yaml", "selinux", "module", i)
+		if module.Name != "" && module.Content != "" {
```

Comment:
Shouldn't this already be checked by validation?

Reply:
Not sure. Discussed with @steven, and validations don't seem to impact this part of the code and we should have some rules here as well.

I can't explain it very well; perhaps Steven can provide some insights about it.

Reply:
Hmm, odd. If I compare to the GRUB code, here's what I see.

We validate e.g. the name here:

https://github.com/coreos/butane/blob/310a1f1bd2725e01f0805e637c4cb39bc36e5763/config/fcos/v1_5/validate.go#L45C1-L47

And here in `translate.go` we just add the name without any validation:

https://github.com/coreos/butane/blob/310a1f1bd2725e01f0805e637c4cb39bc36e5763/config/fcos/v1_5/translate.go#L362C1-L363

Note also the comment there which makes it seem like we run through validation first.

Reply:
If I look at the high-level `TranslateBytes()` function, it does seem like we validate before doing the translation: https://github.com/coreos/butane/blob/310a1f1bd2725e01f0805e637c4cb39bc36e5763/config/util/util.go#L111

Reply:
@yasminvalim yeah @jlebon is certainly correct, sorry about the miss-info. I completely spaced on this. Thank you for the links!

Reply:
Thanks for explain! I changed it!